src/main/java/edu/kis/powp/jobs2d/command/visitor/TransformationScaleVisitorImpl.java

```diff
@@ -0,0 +1,61 @@
+package edu.kis.powp.jobs2d.command.visitor;
+
+import edu.kis.powp.jobs2d.command.*;
+import edu.kis.powp.jobs2d.command.manager.DriverCommandManager;
+
+import java.util.Iterator;
+
+public class TransformationScaleVisitorImpl implements IDriverCommandsVisitor {
+	private int scaleRatio;
+	private int x;
+	private int y;
+	private ComplexCommand complexCommand;
+	private DriverCommandManager driverCommandManager;
+
+	public TransformationScaleVisitorImpl(int scaleRatio, DriverCommandManager commandManager) {
+		this.scaleRatio = scaleRatio;
+		this.complexCommand = new ComplexCommand();
+		this.driverCommandManager = commandManager;
+	}
+
+	private void scale() {
+		this.x = (this.x * this.scaleRatio);
+		this.y = (this.y * this.scaleRatio);
+	}
+
+	@Override
+	public void doForOperateToCommand(OperateToCommand command) {
+		this.x = command.getPosX();
+		this.y = command.getPosY();
+		this.scale();
+		OperateToCommand operateToCommand = new OperateToCommand(this.x, this.y);
+		this.complexCommand.appendCommand(operateToCommand);
+	}
+
+	@Override
+	public void doForSetPositionCommand(SetPositionCommand command) {
+		this.x = command.getPosX();
+		this.y = command.getPosY();
+		this.scale();
+		SetPositionCommand setPositionCommand = new SetPositionCommand(this.x, this.y);
+		this.complexCommand.appendCommand(setPositionCommand);
+	}
+
+	@Override
+	public void doForCompoundCommand(ICompoundCommand command) {
+		Iterator<DriverCommand> iterator = command.iterator();
+		while (iterator.hasNext()) {
+			DriverCommand tempDriverCommand = iterator.next();
+			tempDriverCommand.accept(this);
+		}
```

Comment:
Your solution probably works in many cases, but the correct way should be to create local complex command here, and append it inside this while loop. doForOperateToCommand and doForSetPositionCommand can store their result locally in a new field (e.g., DriverCommand res). This way the logic of visitor methods would be more consistent and atomic.